src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -658,5 +658,17 @@ internal static string GetMemberCallerName(this Symbol member)
                 member.IsExplicitInterfaceImplementation() ? ExplicitInterfaceHelpers.GetMemberNameWithoutInterfaceName(member.Name) :
                 member.Name;
         }
+
+        internal static bool TryGetAccessorSymbolForFieldKeywordBinder(this Symbol symbol, out SourcePropertyAccessorSymbol accessorSymbol)
+        {
+            if (symbol is SourcePropertyAccessorSymbol { Property.IsIndexer: false } accessor)
```

Comment:
I can update `CSharpSemanticModel.GetSpeculativeBinder` to call `AddSpeculativeFieldKeywordBinderIfNeeded` conditionally so that assertion doesn't fail, but this beats its purpose. Is it really worth adding this assertion?

Reply:
Yes I forgot about types and namespaces. The assert can be relaxed accordingly.

> Is it really worth adding this assertion?

The method is meaningful only for certain kinds of symbols. I would add an assertion to verify that the API is used as expected.

Reply:
@AlekseyTs Added two assertions (not sure if they're written in a good way)